packages/runtime/datastore/src/localChannelStorageService.ts

```diff
@@ -13,13 +13,29 @@ export class LocalChannelStorageService implements IChannelStorageService {
     }
 
     public async read(path: string): Promise<string> {
-        const contents = this.readSync(path);
-        return contents !== undefined ? Promise.resolve(contents) : Promise.reject(new Error("Not found"));
+        const blob = await this.readBlob(path);
+        assert(blob !== undefined, "Not Found");
+        assert(blob.contents !== undefined, "Not Found");
+        if (blob.encoding === "base64") {
+            return blob.contents;
+        }
+        return fromUtf8ToBase64(blob.contents);
+    }
+
+    public async readString(path: string): Promise<string> {
+        const blob = await this.readBlob(path);
+        assert(blob !== undefined, "Not Found");
+        assert(blob.contents !== undefined, "Not Found");
+        if (blob.encoding === "base64") {
+            return fromBase64ToUtf8(blob.contents);
+        }
+        return blob.contents;
     }
 
     public async contains(path: string): Promise<boolean> {
-        const contents = this.readSync(path);
-        return contents !== undefined;
+        const blob = await this.readBlob(path);
+        assert(blob !== undefined, "Not Found");
```

Comment:
I do not think this is correct. It should be 
return blob !== undefined;

Reply:
but original readSync is returning blob.contents instead of blob

Reply:
That's correct. But this function tests if blob with given path even exists. It's purpose is to test and return boolean, not to throw if blob does not exist. By adding this assert, you are modifying how this function behaves.

Reply:
oh i see! should I still return blob.contents !== undefined ?

Reply:
nvm i get it now